### DISCUSSION TITLE
Make a function to indent list table correctly

### DISCUSSION THREAD

**Discussion by QiangF** [kjambunathan/org-mode-ox-odt#194] [Ideas]
List table relies on the correct indentation of sub content. A function to format the table is necessary when paste text with incorrect indentation.
For example, when the cursor is at |, we have this list table:

    #+attr_odt: :list-table t
    -
      -
        |

Now yank the following text:

    #+caption: 1
    #+attr_odt: :width 7.1 :style "Caption"
    [[image_file_path]]

The buffer now becomes:

    #+attr_odt: :list-table t
    -
      -
        #+caption: 1
    #+attr_odt: :width 7.1 :style "Caption"
    [[image_file_path]]
    |

When there are a couple of these patterns, it is tedious to fix the indentation by hand.

    #+attr_odt: :list-table t
    -
      -
        #+caption: 1
    #+attr_odt: :width 7.1 :style "Caption"
    [[image_file_path]]
    -
      -
        #+caption: 1
    #+attr_odt: :width 7.1 :style "Caption"
    [[image_file_path]]
    -
      -
        #+caption: 1
    #+attr_odt: :width 7.1 :style "Caption"
    [[image_file_path]]    
    |

The function should format regions between first "-" and the last empty line.

**Comment by kjambunathan** [comment]
Transcluded tables are meant to solve this issue.  Why aren't you using it?

Anyways, use something like this

```lisp
(defun my-yank-at-column ()
  (interactive)
  (let ((ind (current-indentation)))
    (goto-char (point-at-bol))
    (insert
     (with-temp-buffer
       ;; (pop-to-buffer (current-buffer))
       (call-interactively 'yank) ; C-y
       ;; (read-from-minibuffer "YANKED: ")
       (org-do-remove-indentation)
       ;; (read-from-minibuffer "DE-INDENTED: ")
       (indent-rigidly (point-min) (point-max) ind) ; C-x TAB
       ;; (read-from-minibuffer "INDENTED: ")
       (buffer-string)))))
```
You may have to be little bit careful about the leading and trailing newlines.

**Reply by QiangF** [reply]
Transcluded table is not copy and paste friendly. With list table, I don't need to insert a table.

There is a package that does what you suggested.
https://github.com/BohongHuang/indent-yank
Thank you!
